workflows/wf_theiacov_illumina_pe.wdl

```diff
@@ -38,92 +45,130 @@ workflow theiacov_illumina_pe {
+          reference_genome = reference_genome,
+          genome_length = genome_length
+      }
+    call assembly_metrics.stats_n_coverage {
       input:
-      nextclade_tsv = nextclade_one_sample.nextclade_tsv
+        samplename = samplename,
+        bamfile = bwa.sorted_bam
+    }
+    if (organism == "sars-cov-2") {
+      # sars-cov-2 specific tasks
+      call taxon_ID.pangolin4 {
+        input:
+          samplename = samplename,
+          fasta = consensus.consensus_seq
+      }
+      call sc2_calculation.sc2_gene_coverage {
+        input: 
+          samplename = samplename,
+          bamfile = bwa.sorted_bam,
+          min_depth = min_depth
+      }
+    }
+    if (organism == "MPXV" || organism == "sars-cov-2"){ 
+      # tasks specific to either MPXV or sars-cov-2
+      call taxon_ID.nextclade_one_sample {
+        input:
+          genome_fasta = consensus.consensus_seq,
+          dataset_name = select_first([nextclade_dataset_name, organism]),
+          dataset_reference = nextclade_dataset_reference,
+          dataset_tag = nextclade_dataset_tag
+      }
+      call taxon_ID.nextclade_output_parser_one_sample {
+        input:
+        nextclade_tsv = nextclade_one_sample.nextclade_tsv
+      }
     }
+    if (organism == "MPXV" || organism == "sars-cov-2" || organism == "WNV"){ 
+      # tasks specific to MPXV, sars-cov-2, and WNV
+      call ncbi.vadr {
+          input:
+            genome_fasta = consensus.consensus_seq,
+            assembly_length_unambiguous = consensus_qc.number_ATCG
+        }
+      }
+    String assembly_method_nonflu = "~{bwa.bwa_version}; ~{primer_trim.ivar_version}"
   }
-  if (organism == "MPXV" || organism == "sars-cov-2" || organism == "WNV"){ 
-    # tasks specific to MPXV, sars-cov-2, and WNV
-    call ncbi.vadr {
+  if (organism == "flu"){
+    # flu-specific tasks
+    call irma.irma {
       input:
-        genome_fasta = consensus.consensus_seq,
-        assembly_length_unambiguous = consensus_qc.number_ATCG
+        read1 = read_QC_trim.read1_clean,
+        read2 = read_QC_trim.read2_clean,
+        samplename = samplename,
+    }
+    if (defined(irma.irma_assemblies)) {
+      call abricate_flu.abricate_flu {
+        input:
+          assembly = select_first([irma.irma_assembly_fasta]),
+          samplename = samplename,
+          nextclade_flu_h1n1_tag = nextclade_flu_h1n1_tag,
+          nextclade_flu_h3n2_tag = nextclade_flu_h3n2_tag,
+          nextclade_flu_vic_tag = nextclade_flu_vic_tag,
+          nextclade_flu_yam_tag = nextclade_flu_yam_tag,
+      }
+      call consensus_qc_task.consensus_qc as consensus_qc_flu {
+       input:
+       assembly_fasta =  select_first([irma.irma_assembly_fasta]),
+       reference_genome = reference_genome,
+       genome_length = genome_length
+      }
+      if (abricate_flu.run_nextclade && defined(irma.seg4_ha_assembly)) {
+        call taxon_ID.nextclade_one_sample as nextclade_one_sample_flu {
```

Comment:
I don't think we need to call it separately here. If we move the sc2/mpox nextclade tasks below the flu block, and then have an additional if block within it to check if we run it on flu, we could have only one task and then no need for a duplicate.

Reply:
this will also get rid of all of those select_firsts in the output!

Reply:
`if (organism == "MPXV" || organism == "sars-cov-2" || organism == "flu") {`
-> `if select_first([abricate_flu.run_nextclade, "true"]) {`
-> -> `call taxon_ID.nextclade_one_sample {`
-> -> -> `genome_fasta = select_first([consensus.consensus_seq, irma.seg4_ha_assembly]),`
-> -> -> `dataset_name = select_first([nextclade_dataset_name, abricate_flu.nextclade_name, organism]),`
-> -> -> `dataset_reference = select_first([nextclade_dataset_reference, abricate_flu.nextclade_ref]),`
-> -> -> `dataset_tag = select_first([nextclade_dataset_tag, abricate_flu.nextclade_ds_tag])`
->  -> `}`
->  -> ` call taxon_ID.nextclade_output_parser_one_sample { ... }`
 -> `}`
`}`

where `abricate_flu.run_nextclade` is the result of `(abricate_flu.run_nextclade && defined(irma.seg4_ha_assembly))`

Reply:
probably might want to switch around the `select_first([nextclade_dataset_reference, abricate_flu.nextclade_ref])` sections so the abricate_flu comes first in case they accidentally left something in the nextclade fields

Reply:
Oh! Great idea. Makes for much cleaner flow. I can rearrange things and give this a test later today.